### PR TITLE
Supply injector modules based on configuration

### LINKED ISSUE #55: Supply injector with modules based on configuration
The injector object is currently created in `__init__.py` in the `create_app` function. This should be moved to the top level `main.py` where the injector is created based on configuration. For example for storage there will be two modules `LocalStorageModule` and `BlobStorageModule` which are chosen prior to the creation of the injector based on the configuration.

### PR DESCRIPTION
**Pull request by sondreo**
Closes #55.

### CODE REVIEW

File: src/isar/modules.py [hunk 1]
```diff
@@ -131,3 +131,16 @@ def provide_transform(self, map_config_reader: MapConfigReader) -> Transformatio
...
+
+modules: dict = {
+    "coordinate_module": {"default": CoordinateModule},
+    "queues_module": {"default": QueuesModule},
+    "reader_module": {"default": ReaderModule},
+    "request_handler_module": {"default": RequestHandlerModule},
+    "robot_module": {"default": RobotModule},
+    "service_module": {"local": ServiceModule, "echo": ServiceModule},
+    "state_machine_module": {"default": StateMachineModule},
+    "storage_module": {"local": StorageModule, "blob": StorageModule},
```
Comment (aestene) [inline src/isar/modules.py:144]:
```suggestion
    "service_module": {"default": ServiceModule, "local": ServiceModule, "echo": ServiceModule},
    "state_machine_module": {"default": StateMachineModule},
    "storage_module": {"default": ServiceModule, "local": StorageModule, "blob": StorageModule},
```

Comment (aestene) [reply src/isar/modules.py:144]: You should have defaults for the modules that may be interchanged. When they are added with multiple modules the `local` will be the default one for these two for example.

File: main.py [hunk 2]
```diff
@@ -1,9 +1,43 @@
...
+    return injector_modules
+
 
 if __name__ == "__main__":
-    app = create_app()
 
-    host = app.config["HOST"]
-    port = app.config["PORT"]
+    injector_modules: List[Module] = get_injector_modules()
+    injector: Injector = Injector(injector_modules)
```
[the remark this answers is not in the document]
Comment (sondreo) [reply main.py]: Agreed, implemented.

File: src/isar/modules.py (hunk 1 above)
Comment (sondreo) [reply src/isar/modules.py]: Implemented default selections for all modules.

**Pull request merged**